WKPDFView API tests are no longer necessary

aprotyas opened the pull request:
#### 86320e6033bcdd1bfeb96d10e6647c2358e24248
<pre>
WKPDFView API tests are no longer necessary
<a href="https://bugs.webkit.org/show_bug.cgi?id=296653">https://bugs.webkit.org/show_bug.cgi?id=296653</a>
<a href="https://rdar.apple.com/157052720">rdar://157052720</a>

Reviewed by Aditya Keerthi.

This patch is a prequel to WKPDFView removal (c.f. webkit.org/b/296654).

WKPDFView is no longer the default PDF configuration on trunk. In this
patch, we do a couple of things:

- Migrate (most) tests from WKPDFView.mm to the UnifiedPDF test suite.
- Remove WKPDFView-specific tests.
- Remove WKPDFView.mm and TestPDFHostViewController, which was written
  as test support for WKPDFView.
- Remove WKPDFView-specific FindInPage API tests.
- Remove the UnifiedPDF=OFF variant of LoadPDFWithSandboxCSPDirective.
  Again, since test coverage for the OFF config is not necessary moving
  forward. This allows us to remove ContentSecurityPolicyTestHelpers.
- Adjust the NavigatingFromPDFDoesNotLeaveWebViewInactive API test to
  not stand up a swizzled PDFHostViewController, and instead use a
  UnifiedPDFPlugin instead.

* Tools/TestWebKitAPI/SourcesCocoa.txt:
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/ContentSecurityPolicy.mm:
(TEST(ContentSecurityPolicy, InvalidRequireTrustedTypesFor)):
(TEST(ContentSecurityPolicy, LoadPDFWithSandboxCSPDirective)): Deleted.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/ContentSecurityPolicyTestHelpers.h: Removed.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/ContentSecurityPolicyTestHelpers.mm: Removed.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/FindInPage.mm:
(swizzledPerformTextSearchWithQueryString): Deleted.
(TEST(WebKit, FindInPDF)): Deleted.
(TEST(WebKit, FindInPDFAfterReload)): Deleted.
(TEST(WebKit, FindInPDFAfterFindInPage)): Deleted.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/UnifiedPDFTests.mm:
(-[PDFPrintUIDelegate _webView:printFrame:pdfFirstPageSize:completionHandler:]):
(-[PDFPrintUIDelegate waitForPageSize]):
(-[PDFPrintUIDelegate lastPrintedFrame]):
(TestWebKitAPI::testPDFData):
(TestWebKitAPI::UNIFIED_PDF_TEST):
(TestWebKitAPI::checkFrame):
(TestWebKitAPI::isBackground):
(TestWebKitAPI::pdfServerWithSandboxCSPDirective):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKPDFView.mm: Removed.
* Tools/TestWebKitAPI/Tests/ios/ApplicationStateTracking.mm:
(TestWebKitAPI::TEST(ApplicationStateTracking, NavigatingFromPDFDoesNotLeaveWebViewInactive)):
* Tools/TestWebKitAPI/ios/TestPDFHostViewController.h: Removed.
* Tools/TestWebKitAPI/ios/TestPDFHostViewController.mm: Removed.

Canonical link: <a href="https://commits.webkit.org/298023@main">https://commits.webkit.org/298023@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d9b5858a6b19f8879b25985541c423b10ef097cc

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/113927 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/33679 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/24137 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/120093 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/64706 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/d0ec4958-1599-4a96-9557-f17507a20650) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/115816 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/34307 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/42240 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/86581 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/41624 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/307fc3bc-502d-4496-8788-6c80a205e446) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/116875 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/27302 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/102319 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/66959 "Passed tests") | | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/26499 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/20449 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/63803 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/96670 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/20566 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/123320 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/40964 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/30501 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/95419 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/41338 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/98531 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/95194 "Found 100 new API test failures: /TestWebKit:WebKit.EvaluateJavaScriptThatCreatesBlob, /TestWebKit:WebKit.DOMWindowExtensionCrashOnReload, /TestWebKit:WebKit2TextFieldBeginAndEditEditingTest.TextFieldDidBeginAndEndEditingEvents, /TestWebKit:WebKit2TextFieldBeginAndEditEditingTest.TextFieldDidEndShouldBeDispatchedForRemovedFocusField, /TestWebKit:WebKit.UserMediaBasic, /WebKitGTK/TestWebKitPolicyClient:/webkit/WebKitPolicyClient/autoplay-policy, /WebKitGTK/TestWebViewEditor:/webkit/WebKitWebView/editor-state/typing-attributes, /TestWebKit:WebKit.FirstMeaningfulPaint, /TestWebKit:WebKit.FrameMIMETypeHTML, /WebKitGTK/TestWebKitWebView:/webkit/WebKitWebView/is-playing-audio ... (failure)") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/40332 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/18136 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/37100 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/18271 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/40838 "Built successfully") | | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/40471 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/43771 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/42228 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/26f947111f8de17fc67a20020950e1ab2a8bcb5c. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48683

aprotyas committed:
WKPDFView API tests are no longer necessary
https://bugs.webkit.org/show_bug.cgi?id=296653
rdar://157052720

Reviewed by Aditya Keerthi.

This patch is a prequel to WKPDFView removal (c.f. webkit.org/b/296654).

WKPDFView is no longer the default PDF configuration on trunk. In this
patch, we do a couple of things:

- Migrate (most) tests from WKPDFView.mm to the UnifiedPDF test suite.
- Remove WKPDFView-specific tests.
- Remove WKPDFView.mm and TestPDFHostViewController, which was written
  as test support for WKPDFView.
- Remove WKPDFView-specific FindInPage API tests.
- Remove the UnifiedPDF=OFF variant of LoadPDFWithSandboxCSPDirective.
  Again, since test coverage for the OFF config is not necessary moving
  forward. This allows us to remove ContentSecurityPolicyTestHelpers.
- Adjust the NavigatingFromPDFDoesNotLeaveWebViewInactive API test to
  not stand up a swizzled PDFHostViewController, and instead use a
  UnifiedPDFPlugin instead.

* Tools/TestWebKitAPI/SourcesCocoa.txt:
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/ContentSecurityPolicy.mm:
(TEST(ContentSecurityPolicy, InvalidRequireTrustedTypesFor)):
(TEST(ContentSecurityPolicy, LoadPDFWithSandboxCSPDirective)): Deleted.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/ContentSecurityPolicyTestHelpers.h: Removed.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/ContentSecurityPolicyTestHelpers.mm: Removed.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/FindInPage.mm:
(swizzledPerformTextSearchWithQueryString): Deleted.
(TEST(WebKit, FindInPDF)): Deleted.
(TEST(WebKit, FindInPDFAfterReload)): Deleted.
(TEST(WebKit, FindInPDFAfterFindInPage)): Deleted.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/UnifiedPDFTests.mm:
(-[PDFPrintUIDelegate _webView:printFrame:pdfFirstPageSize:completionHandler:]):
(-[PDFPrintUIDelegate waitForPageSize]):
(-[PDFPrintUIDelegate lastPrintedFrame]):
(TestWebKitAPI::testPDFData):
(TestWebKitAPI::UNIFIED_PDF_TEST):
(TestWebKitAPI::checkFrame):
(TestWebKitAPI::isBackground):
(TestWebKitAPI::pdfServerWithSandboxCSPDirective):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKPDFView.mm: Removed.
* Tools/TestWebKitAPI/Tests/ios/ApplicationStateTracking.mm:
(TestWebKitAPI::TEST(ApplicationStateTracking, NavigatingFromPDFDoesNotLeaveWebViewInactive)):
* Tools/TestWebKitAPI/ios/TestPDFHostViewController.h: Removed.
* Tools/TestWebKitAPI/ios/TestPDFHostViewController.mm: Removed.

Canonical link: https://commits.webkit.org/298023@main

webkit-commit-queue commented:
Committed 298023@main (86320e6033bc): <https://commits.webkit.org/298023@main>

Reviewed commits have been landed. Closing PR #48683 and removing active labels.
